acegen/models/__init__.py

```diff
@@ -41,37 +42,50 @@ def extract(path):
         return path
 
 
-models = {
-    "gru": (
+def gru_model_factory(*args, **kwargs):
+    return (
         create_gru_actor,
         create_gru_critic,
         create_gru_actor_critic,
         resources.files("acegen.priors") / "chembl_filtered_vocabulary.txt",
         resources.files("acegen.priors") / "gru_chembl_filtered.ckpt",
         SMILESTokenizer(),
-    ),
-    "gru_guacamol": (
-        create_gru_actor,
-        create_gru_critic,
-        create_gru_actor_critic,
-        resources.files("acegen.priors") / "gru_guacamol_vocabulary.ckpt",
-        resources.files("acegen.priors") / "gru_guacamol.ckpt",
-        SMILESTokenizer3(),
-    ),
-    "lstm": (
+    )
+
+
+def lstm_model_factory(*args, **kwargs):
+    return (
         create_lstm_actor,
         create_lstm_critic,
         create_lstm_actor_critic,
         resources.files("acegen.priors") / "chembl_vocabulary.txt",
         resources.files("acegen.priors") / "lstm_chembl.ckpt",
         SMILESTokenizer(),
-    ),
-    "gpt2": (
+    )
+
+
+def gpt2_model_factory(*args, **kwargs):
+    return (
         create_gpt2_actor,
         create_gpt2_critic,
         create_gpt2_actor_critic,
         resources.files("acegen.priors") / "enamine_real_vocabulary.txt",
         extract(resources.files("acegen.priors") / "gpt2_enamine_real.ckpt"),
         SMILESTokenizer2(),
-    ),
+    )
+
+
+# Default models
+models = {
+    "gru": gru_model_factory,
+    "lstm": lstm_model_factory,
+    "gpt2": gpt2_model_factory,
 }
+
+
+def register_model(name, factory):
+    """Register a model factory."""
```

Comment:
I have modified the tutorials https://github.com/Acellera/acegen-open/pull/25/commits/f27daf1ce1ae43d1e362c92d3e8aad12ea8a6bc3. I have not changed the namings because I feel it is easier to understand as it is. I feel more natural to define in the context `model: model_name` rather than factory... And when you register a model you register it with its factory, so still seems ok to call it `register_model`